src/interface.ts

```diff
@@ -7,18 +7,18 @@ export interface Balances {
 }
 
 export interface IWallet {
-  getBalances(lock?): Promise<Balances>
+  getBalances(): Promise<Balances>
```

Comment:
there might be some issue with the type here. maybe IWallet as a type is not being used, but getBalances may require a lock when it's called within a thread that have already acquired the lock.

here is why (from a design perspective at least):
- if a user call `getBalances()` directly, then no lock is pass to the function, and getBalances will acquire the lock by itself.
- a function like `pay()` will need to know the balance of the user during the payment process. the lock would already been acquired prior to `getBalance()`, so it's passing it `lock` downstream to `getBalance()`

Reply:
`getBalances` in `UserWallet` might not be type correctly? (`IWallet`?)
those types has not been well maintained and would need more attention!